vignettes/ChromBand.Rmd

```diff
@@ -0,0 +1,379 @@
+
+</figure>
+
+We fit a factorial model to see whether the distribution of $y_i$ is associated with category membership. Specifically, for category $j$, we fit the model $$\label{eq:lm1}  y_i = \beta_0 + \beta_1 a_{ij} +\varepsilon_{i}$$
+where $a_{ij} = 1$ if gene $i$ belongs to category $j$, and $0$ otherwise. The index $i$ may range over the full gene universe, or a subset, depending on whether one wishes to perform global or local tests. The null hypothesis of no association is represented by $H_0: \beta_1 = 0$. The model nominally assumes that the $y_i$-s are Normally distributed with equal variance, but in practice the results are robust against mild deviations. The presence of an intercept term allows nonzero overall mean, which can be important in many situations, especially for local tests. We expect the test to be fairly insensitive to distributional assumptions on the $y_i$-s.
+
+We can fit (1) by least squares and test $H_0: \beta_1 = 0$ to obtain a marginal test for each category $j$; in this case, each chromosome band. The procedure also generalizes to incorporate the nesting structure of chromosome bands. Specifically, if band $j_2$ (e.g., 12q21.1) is nested within a coarser band $j_1$ (e.g., 12q21) and the more granular band $j_2$ is significant, then the effect of membership in $j_1$ over and above the effect attributable to membership in $j_2$ can be tested by fitting the model $$\label{eq:lm2}
+  y_i = \beta_0 + \beta_1 a_{ij_1} + \beta_2 a_{ij_2} + \varepsilon_{i}$$ and testing the null hypothesis $H_0: \beta_1 = 0$. Multiple significant sub-bands and multiple levels of nesting can be incorporated by including further terms in the model. The complete process can be summarized as follows: Start by marginally testing each band which has no sub-bands. For all other bands, first test all sub-bands, then test the current band using a linear model that includes a term for each *significant* sub-band.
+
+We apply this procedure to perform global tests using per-gene $t$-statistics as a measure of differential expression in BCR/ABL relative to NEG samples. As with the Hypergeometric tests, we start by creating objects of class *"ChrMapLinearMParams"*.
+
+```{r LMtestSetup}
+
+params <- new("ChrMapLinearMParams",
+              conditional = FALSE,
+              testDirection = "up",
+              universeGeneIds = entrezUniverse,
+              geneStats = tstats,
+              annotation = "hgu95av2",
+              pvalueCutoff = 0.01, 
+              minSize = 4L)
+
+params@graph <- makeChrBandGraph(params@annotation, params@universeGeneIds)
+params@gsc <- makeChrBandGSC(params@graph)
+paramsCond <- params
+paramsCond@conditional <- TRUE
+```
+
+The tests are performed, and the results summarized by
+
+```{r LMtest}
+lmans <- linearMTest(params)
+lmansCond <- linearMTest(paramsCond)
+chrSortOrder(summary(lmans))
+
+chrSortOrder(summary(lmansCond))
+##
```

Comment:
Remove blank line 351 and 353 (I know this is in the PDF, but it doesn't need to be there).